kiwi/storage/subformat/vagrant_base.py

```diff
@@ -0,0 +1,304 @@
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import json
+import os
+import random
+from textwrap import dedent
+from tempfile import mkdtemp
+from string import Template
+
+from kiwi.storage.subformat.base import DiskFormatBase
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiFormatSetupError
+)
+
+
+class VagrantConfigTemplate(object):
+    """
+    **Generate a Vagrantfile configuration template**
+
+    This class creates a simple template for the Vagrantfile that is included
+    inside a vagrant box.
+
+    The included Vagrantfile carries additional information for vagrant: by
+    default that is just the boxes' MAC address, but depending on the provider
+    additional information need to be present. These can be passed via the
+    parameter ``custom_settings`` to the method :meth:`get_template`.
+
+    Example usage:
+
+    The default without any additional settings will result in this
+    Vagrantfile:
+
+    .. code:: python
+
+        >>> vagrant_config = VagrantConfigTemplate()
+        >>> print(
+        ...     vagrant_config.get_template()
+        ...     .substitute({'mac_address': 'deadbeef'})
+        ... )
+        Vagrant.configure("2") do |config|
+          config.vm.base_mac = "deadbeef"
+        end
+
+    If your provider/box requires additional settings, provide them as follows:
+
+    .. code:: python
+
+        >>> extra_settings = dedent('''
+        ... config.vm.hostname = "no-dead-beef"
+        ... config.vm.provider :special do |special|
+        ...   special.secret_settings = "please_work"
+        ... end
+        ... ''').strip()
+        >>> print(
+        ...     vagrant_config.get_template(extra_settings)
+        ...     .substitute({'mac_address': 'DEADBEEF'})
+        ... )
+        Vagrant.configure("2") do |config|
+          config.vm.base_mac = "DEADBEEF"
+          config.vm.hostname = "no-dead-beef"
+          config.vm.provider :special do |special|
+            special.secret_settings = "please_work"
+          end
+        end
+    """
+
+    _INDENT = '  '
+
+    _HEADER = dedent('''
+        Vagrant.configure("2") do |config|
+    ''').strip() + os.linesep
+
+    _MAC = _INDENT + dedent('''
+        config.vm.base_mac = "${mac_address}"
+    ''').strip() + os.linesep
+
+    _END = dedent('''
+        end
+    ''').strip()
```

Comment:
I usually avoid class global values especially when we create an instance object of the class. The template is created in code later as ```template = VagrantConfigTemplate()``` and we operate on template. Thus all this values from above _INDENT, _HEADER, etc... deserves a constructor and should imho be set in the __init__ method.

Reply:
I'll change that.

I have used class global values since they will be constructed once the module is imported and not every time a template is created. But since it's unlikely that we will create two templates, it won't make a difference.